src/adam7.rs

```diff
@@ -233,6 +266,145 @@ pub fn expand_pass(
     }
 }
 
+// TODO: Export this function in lib.rs.
+// Should we have a separate function, or can we use an enum to determine whether
+// to use default or splat?
+//
+// |frame_width_in_pixels| is only needed for bit splat expand, but currently it's
+// required as a parameter for both splat expand types. Would it be possible to
+// pass the width only when it's a bit splat expand? Or is there an alternative
+// way to retrieve the width without passing it explicitly?
+//
+// The `splat_expand_pass` fills pixels in a splatted way, unlike `expand_pass`
+// which only writes to specific required positions (byte or bit). It splats the
+// pixel value without leaving any pixels empty, helping to produce smoother
+// animation for interlaced images.
+// For example, after the 3rd pass using expand_pass, the image for an 8x8 size
+// would look like this:
+//   1 - - - 2 - - -
+//   - - - - - - - -
+//   - - - - - - - -
+//   - - - - - - - -
+//   3 - - - 3 - - -
+//   - - - - - - - -
+//   - - - - - - - -
+//   - - - - - - - -
+// On the other hand, after the 3rd pass using splat_expand_pass, the image
+// would look like this:
+//   1 1 1 1 2 2 2 2
+//   1 1 1 1 2 2 2 2
+//   1 1 1 1 2 2 2 2
+//   1 1 1 1 2 2 2 2
+//   3 3 3 3 3 3 3 3
+//   3 3 3 3 3 3 3 3
+//   3 3 3 3 3 3 3 3
+//   3 3 3 3 3 3 3 3
+pub fn splat_expand_pass(
+    img: &mut [u8],
+    img_row_stride: usize,
+    interlaced_row: &[u8],
+    interlace_info: &Adam7Info,
+    bits_per_pixel: u8,
+    frame_width_in_pixels: usize,
+) {
+    let bits_pp = bits_per_pixel as usize;
+
+    let bit_indices = expand_adam7_bits(img_row_stride, interlace_info, bits_pp);
+
+    if bits_pp < 8 {
+        for (pos, px) in bit_indices.zip(subbyte_pixels(interlaced_row, bits_pp)) {
+            bit_splat_expand(
+                img,
+                img_row_stride * 8,
+                px,
+                interlace_info,
+                bits_pp,
+                pos,
+                frame_width_in_pixels * bits_pp,
+            );
+        }
+    } else {
+        let bytes_pp = bits_pp / 8;
+
+        for (bitpos, px) in bit_indices.zip(interlaced_row.chunks(bytes_pp)) {
+            let start_byte = bitpos / 8;
+            byte_splat_expand(
+                img,
```

Comment:
Would it make sense to call this with `&mut img[start_byte...]` and then not pass `start_byte` as a separate argument?